### PR TITLE
Add OpenSSL related settings to CUSTOM_COMPILER_ENV_VARS

### PR DESCRIPTION
**Pull request by a7ehuo**
`JITServer` needs to be built with `libssl`. By default, it
is built against the `OpenSSL` installed on the build machine.
The same as when `--enable-jitserver --with-openssl=system`
is specified.

This change passes the `OpenSSL` build flags from OpenJDK to
JIT build environment.
- Add `OPENSSL_DIR`, `OPENSSL_CFLAGS`, and `OPENSSL_BUNDLE_LIB_PATH`
  to `CUSTOM_COMPILER_ENV_VARS` and `CMAKE_ARGS`.
- Bundle `libssl` if `--with-openssl=fetched --enable-openssl-bundling`
  is set.

Examples on the supported flags:
--enable-jitserver --with-openssl=fetched
--enable-jitserver --with-openssl=fetched --enable-openssl-bundling
--enable-jitserver --with-openssl=system

The JIT build change is addressed in:
eclipse/openj9#5789, eclipse/openj9#7846

This is a similar change as ibmruntimes/openj9-openjdk-jdk8#303

Issue: eclipse/openj9#5720

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@keithc-ca All comments addressed. Ready for another review. Thanks.
@mpirvu FYI.

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -450,6 +450,18 @@ endif
...
+
+    ifneq (,$(OPENSSL_DIR))
+      CUSTOM_COMPILER_ENV_VARS += OPENSSL_DIR="$(OPENSSL_DIR)"
+    endif
+
+    ifneq (,$(OPENSSL_BUNDLE_LIB_PATH))
+      CUSTOM_COMPILER_ENV_VARS += OPENSSL_BUNDLE_LIB_PATH="$(OPENSSL_BUNDLE_LIB_PATH)"
+    endif
   endif
 endif
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: Please add comments to these `endif` lines identifying the conditions they close.

File: closed/make/copy/Copy-java.base.gmk [hunk 2]
```diff
@@ -87,5 +87,27 @@ ifneq ($(OPENSSL_BUNDLE_LIB_PATH), )
...
+
+    ifneq ($(OPENSSL_SSL_LIB_NAME), )
+      OPENSSL_SSL_TARGET_LIB = $(LIB_DST_DIR)/$(OPENSSL_SSL_LIB_NAME)
+      TARGETS += $(OPENSSL_SSL_TARGET_LIB)
+      $(OPENSSL_SSL_TARGET_LIB): $(OPENSSL_BUNDLE_LIB_PATH)/$(OPENSSL_SSL_LIB_NAME)
+		@$(MKDIR) -p $(@D)
+		$(CP) $< $@
+    endif
+  endif
 endif
```
Comment (keithc-ca) [inline closed/make/copy/Copy-java.base.gmk]: Same here.

**Comment by keithc-ca** [comment]
Jenkins compile xlinux,xlinuxjit jdknext

**Pull request merged**